glib/src/error.rs

```diff
@@ -152,49 +152,55 @@ pub trait ErrorDomain: Copy {
 /// Generic error used for functions that fail without any further information
 #[macro_export]
 macro_rules! bool_error(
-// Plain strings
-    ($msg:expr) => {{
-        $crate::BoolError::new(
-            $msg,
-            file!(),
-            $crate::function_name!(),
-            line!(),
-        )
-    }};
-
-// Format strings
     ($($msg:tt)*) =>  {{
-        $crate::BoolError::new(
-            format!($($msg)*),
-            file!(),
-            $crate::function_name!(),
-            line!(),
-        )
+        match ::std::format_args!($($msg)*) {
```

Comment:
The `match` because the return value of `format_args!` is otherwise annoying to work with?

Reply:
Yeah, I tried storing it in a normal variable using `let` and that caused borrow checker issues

Reply:
Nicer than my workaround for that :) I usually use a closure